scenarios/ecs_privesc_evade_protection/terraform/push-dockerfile.py

```diff
@@ -0,0 +1,76 @@
+# How to run:
+# python3 push-dockerfile.py
+import boto3
+import subprocess
+import base64
+import argparse
+import time
+
+
+def run_command_with_retry(command, max_retries=3, delay=5):
+    for attempt in range(max_retries):
+        try:
+            subprocess.run(command, shell=True, check=True)
+            return
+        except subprocess.CalledProcessError:
+            if attempt < max_retries - 1:
+                print(f"Attempt `{command}` failed. Retrying after {delay} seconds...{attempt + 1}/{max_retries + 1}")
+                time.sleep(delay)
+
+
+def create_ecr_repository(client, repository_name):
+    try:
+        response = client.describe_repositories(repositoryNames=[repository_name])
+        print(f"Repository {repository_name} already exists.")
+        return response["repositories"][0]["repositoryUri"]
+    except client.exceptions.RepositoryNotFoundException:
+        print(f"Creating repository {repository_name}.")
+        response = client.create_repository(repositoryName=repository_name)
+        return response["repository"]["repositoryUri"]
+
+
+def get_docker_login_cmd(client, region):
+    token = client.get_authorization_token()["authorizationData"][0]
+    username, password = (
+        base64.b64decode(token["authorizationToken"]).decode().split(":")
+    )
+    registry = token["proxyEndpoint"]
+    return f"docker login --username {username} --password {password} {registry}"
+
+
+def docker_build_and_push(repository_uri, image_tag, path):
+    # Build the Docker image
+    subprocess.run(f"docker build -t {repository_uri}:{image_tag} {path}", shell=True, check=True)
```

Comment:
Add `--platform linux/amd64` to the build subprocess, I'm on a apple silicon machine which is based on arm

Logs from the ecs server

```bash
[root@ip-192-168-150-41 ~]# docker logs 6d265c0e1c35
exec /usr/local/bin/docker-php-entrypoint: exec format error
```